### PR TITLE
implement vsplit operation for all backends

### PR DESCRIPTION
**Pull request by Shi-pra-19**
Added new `vsplit` ops, which split a tensor into multiple sub-tensors vertically. 
Supported across all backends (NumPy, TensorFlow, Torch, JAX, OpenVINO).

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -7194,6 +7194,98 @@ def vstack(xs):
...
+            end = all_indices[i + 1]
+            if start is None or end is None:
+                output_size = None
+            else:
+                output_size = end - start
+            output_shape = list(x_shape)
+            output_shape[0] = output_size
+            outputs.append(KerasTensor(output_shape, dtype=x.dtype))
+
+        return outputs
```
[the remark this answers is not in the document]
Comment (Shi-pra-19) [reply keras/src/ops/numpy.py]: Done

File: keras/src/backend/tensorflow/numpy.py [hunk 2]
```diff
@@ -3009,6 +3009,10 @@ def vstack(xs):
     return tf.concat(xs, axis=0)
 
 
+def vsplit(x, indices_or_sections):
+    return tf.experimental.numpy.vsplit(x, indices_or_sections)
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: Actually, I would prefer you do `return split(x, indices_or_sections, axis=0)`.

Early on, we tried using `tf.experimental.numpy` but ran into issues.

File: keras/src/ops/numpy_test.py [hunk 3]
```diff
@@ -2608,6 +2616,14 @@ def test_split(self):
         with self.assertRaises(ValueError):
             knp.split(x, 2, axis=1)
 
+    def test_vsplit(self):
+        x = KerasTensor((2, 3))
+        self.assertEqual(knp.vsplit(x, 2)[0].shape, (1, 3))
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py]: Can you also do `self.assertEqual(knp.vsplit(x, 2)[1].shape, (1, 3))`

File: keras/src/ops/numpy_test.py [hunk 4]
```diff
@@ -2608,6 +2616,14 @@ def test_split(self):
...
+        self.assertEqual(len(knp.vsplit(x, [1, 2])), 3)
+        self.assertEqual(knp.vsplit(x, [1, 2])[0].shape, (1, 3))
+        self.assertEqual(knp.vsplit(x, [1, 2])[1].shape, (1, 3))
+        self.assertEqual(knp.vsplit(x, [1, 2])[2].shape, (0, 3))
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py]: Tensors of 0 size (i.e. where any dimension is 0) are generally poorly supported. Can you change the example to use (5, 3) as the input shape for instance?

File: keras/src/ops/numpy_test.py [hunk 5]
```diff
@@ -5291,6 +5307,34 @@ def fn(x, indices, axis):
...
+
+        # test indices_or_sections as tensor
+        x = knp.array([[1, 2, 3], [3, 2, 1]])
+        indices_or_sections = knp.array([1])
+        x_np = np.array([[1, 2, 3], [3, 2, 1]])
+        indices_or_sections_np = np.array([1])
+        self.assertAllClose(
+            knp.vsplit(x, indices_or_sections),
+            np.vsplit(x_np, indices_or_sections_np),
+        )
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py]: For the indices split examples, can you use a slightly more complicated examples with several indices instead of just one?

File: keras/src/backend/jax/numpy.py [hunk 6]
```diff
@@ -1350,6 +1350,10 @@ def vstack(xs):
     return jnp.vstack(xs)
 
 
+def vsplit(x, indices_or_sections):
+    return jnp.vsplit(x, indices_or_sections)
```
Comment (hertschuh) [inline keras/src/backend/jax/numpy.py:1359]: Add `x = convert_to_tensor(x)`.

**Review by hertschuh — COMMENTED** [review]
Thanks for the adding this!

**Comment by hertschuh** [comment]
You'll need to rebase too.

File: keras/src/backend/tensorflow/numpy.py (hunk 2 above)
Comment (Shi-pra-19) [reply keras/src/backend/tensorflow/numpy.py]: Updated 👍🏻

File: keras/src/backend/jax/numpy.py (hunk 6 above)
Comment (Shi-pra-19) [reply keras/src/backend/jax/numpy.py:1359]: Updated 👍🏻

**Comment by Shi-pra-19** [comment]
I’ve refactored and improved the tests as suggested and rebased on the latest master. Looking forward to your feedback!

File: keras/src/ops/numpy_test.py (hunk 5 above)
Comment (Shi-pra-19) [reply keras/src/ops/numpy_test.py]: Updated

File: keras/src/ops/numpy_test.py (hunk 4 above)
Comment (Shi-pra-19) [reply keras/src/ops/numpy_test.py]: Updated

File: keras/src/ops/numpy_test.py (hunk 3 above)
Comment (Shi-pra-19) [reply keras/src/ops/numpy_test.py]: Updated

**Review by hertschuh — APPROVED** [review]
Thanks!
